### ISSUE TITLE
Convergence issue with time dependent models

### ISSUE THREAD

**Issue by smaret** [smaret/astrochem#63]
This [time dependent model](https://gist.github.com/smaret/8cd984d8c5079b0412b2fc89cabb46a4) has a jump in temperature around t = 10 000 years. Other parameters are constant. 

```
% ~/Models/astrochem/src/astrochem -v input_8.ini 
Reading reactions network from full_original.chm... Found 6774 reactions involving 650 species.
astrochem: warning: use of a input grain abundance is deprecated, yet still supported. Please use the grain_gas_mass_ratio and grain_mass_density parameters instead.
Computing abundances in cell 0...
t = 1.00e-06  delta_t = 1.45e-07
t = 1.31e-06  delta_t = 2.40e-07
t = 1.72e-06  delta_t = 2.40e-07
t = 2.26e-06  delta_t = 2.40e-07
t = 2.97e-06  delta_t = 4.47e-07
t = 3.90e-06  delta_t = 3.20e-07
[...]
t = 1.11e+04  delta_t = 1.89e+02
t = 1.11e+04  delta_t = 1.89e+02
t = 1.11e+04  delta_t = 1.89e+02

[CVODE ERROR]  CVode
  At t = 3.49047e+11 and h = 5.83476, the error test failed repeatedly or with |h| = hmin.

t = 1.11e+04  delta_t = 1.89e+02

[CVODE ERROR]  CVode
  At t = 3.49047e+11 and h = 4.85109e-06, the error test failed repeatedly or with |h| = hmin.
[...]
[CVODE WARNING]  CVode
  The above warning has been issued mxhnil times and will not be issued again for this problem.

t = 1.11e+04  delta_t = 8.92e-02
t = 1.11e+04  delta_t = 2.72e-01
[...]
t = 1.12e+04  delta_t = 3.50e+00
t = 1.12e+04  delta_t = 3.50e+00
Done with cell 0.
```

The convergence issue occurs at the time of the jump.

**Comment by smaret** [comment]
This is likely doe to a discontinuity in the RHS of the ODE system: the time step should be re-adjusted just after the jump in temperature, but it is not (`delta_t = 1.89e+02`). In case of discontinuities, the `CVode` documentation suggests to re-initialise the ODE solver with the `CVodeReInit` function:

> One important use of the` CVodeReInit` function is in the treating of jump discontinuities in the RHS function. Except in cases of fairly small jumps, it is usually more efficient to stop at each point of discontinuity and restart the integrator with a readjusted ODE model, using a call to `CVodeReInit`

**Comment by smaret** [comment]
The convergence error is fixed with a0b5774. However, there are still some warnings at the time of the jump:

```
t = 8.34e+03  delta_t = 1.66e+02
t = 1.09e+04  delta_t = 2.57e+02

[CVODE WARNING]  CVode
  Internal t = 3.43742e+11 and h = 9.54626e-07 are such that t + h = t on the next step. The solver will continue anyway.
```

I think this is because when the jump occurs, the time step is adjusted to a small value (`h = 10e-6` s) at a late time (`t = 3e+11` s). Therefore at the next time step, `t + h` is almost equal to `t`. The solver recovers after a few step though, so the warning can probably be ignored.
